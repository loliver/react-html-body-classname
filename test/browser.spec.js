import React from 'react'
import { render } from '@testing-library/react'

import { HtmlClassName, BodyClassName } from '../'

describe('HtmlClassName (in a browser)', () => {
  global.beforeEach(() => {
    HtmlClassName.canUseDOM = true
    global.document.documentElement.className = ''
  })

  it('changes the document html class name on mount', () => {
    var className = 'hello world'
    render(React.createElement(HtmlClassName, { className: className }))
    expect(global.document.documentElement.className).toEqual(className)
  })

  it('does not erase existing html class names', () => {
    global.document.documentElement.className = 'testing'
    var className = 'hello world'
    render(React.createElement(HtmlClassName, { className: className }))
    expect(global.document.documentElement.className).toEqual(
      'testing hello world'
    )
  })

  it('does not erase and or duplicate existing html class names', () => {
    global.document.documentElement.className = 'testing hello'
    var className = 'hello world'
    render(React.createElement(HtmlClassName, { className: className }))
    expect(global.document.documentElement.className).toEqual(
      'testing hello world'
    )
  })

  it('supports nesting, gathering all classNames used', () => {
    var called = jest.fn()
    var firstName = 'hello world'
    var secondName = 'foo bar'

    const Component1 = () => <HtmlClassName className={firstName} />
    const Component2 = () => {
      called()

      return (
        <HtmlClassName className={secondName}>
          <div>
            <Component1 />
          </div>
        </HtmlClassName>
      )
    }

    const { baseElement } = render(<Component2 />)

    expect(called).toHaveBeenCalled()
    expect(baseElement).toMatchSnapshot()
    expect(global.document.documentElement.classList).toContain('hello')
    expect(global.document.documentElement.classList).toContain('world')
    expect(global.document.documentElement.classList).toContain('foo')
    expect(global.document.documentElement.classList).toContain('bar')
  })
})

describe('BodyClassName (in a browser)', () => {
  beforeEach(() => {
    BodyClassName.canUseDOM = true
    global.document.body.className = ''
  })

  it('changes the document body class name on mount', () => {
    var className = 'hello world'
    render(<BodyClassName className={className} />)
    expect(global.document.body.className).toEqual(className)
  })

  it('does not erase existing body class names', () => {
    global.document.body.className = 'testing'
    var className = 'hello world'
    render(<BodyClassName className={className} />)
    expect(global.document.body.className).toEqual('testing hello world')
  })

  it('does not erase and or duplicate existing body class names', () => {
    global.document.body.className = 'testing hello'
    var className = 'hello world'
    render(<BodyClassName className={className} />)
    expect(global.document.body.className).toEqual('testing hello world')
  })

  it('supports nesting, gathering all classNames used', () => {
    var called = jest.fn()
    var firstName = 'hello world'
    var secondName = 'foo bar'

    const Component1 = () => <BodyClassName className={firstName} />
    const Component2 = () => {
      called()

      return (
        <BodyClassName className={secondName}>
          <div>
            <Component1 />
          </div>
        </BodyClassName>
      )
    }

    const { baseElement } = render(<Component2 />)

    expect(called).toHaveBeenCalled()
    expect(baseElement).toMatchSnapshot()
    expect(global.document.body.classList).toContain('hello')
    expect(global.document.body.classList).toContain('world')
    expect(global.document.body.classList).toContain('foo')
    expect(global.document.body.classList).toContain('bar')
  })
})
