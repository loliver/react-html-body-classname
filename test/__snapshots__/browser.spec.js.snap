// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BodyClassName (in a browser) supports nesting, gathering all classNames used 1`] = `
<body
  class="hello world foo bar"
>
  <div>
    <div />
  </div>
</body>
`;

exports[`HtmlClassName (in a browser) supports nesting, gathering all classNames used 1`] = `
<body>
  <div>
    <div />
  </div>
</body>
`;
